packages/components/src/components/Address/Address.tsx

```diff
@@ -23,46 +27,47 @@ export interface AddressProps {
    */
   shortened?: boolean;
   /**
-   * RPC URL for provider, required for ens lookup
+   * Set to true for ENS lookup
    */
-  rpcURL?: string;
+  ens?: boolean;
 }
 
 /**
  * A component to display an address
  */
 export const Address: React.FC<AddressProps> = ({
   value,
+  provider,
   copiable = false,
   shortened = false,
-  rpcURL,
+  ens = false,
 }) => {
-  const [error, setError] = useState<null | string>(null);
+  const [error, setError] = useState<undefined | string>(undefined);
   const [copied, setCopied] = useState<boolean>(false);
   let feedbackTimeOut: ReturnType<typeof setTimeout>;
   let displayAddress: string = value || '';
-  const [ens, setEns] = useState<string | null | undefined>(null);
-  const provider: ethers.providers.JsonRpcProvider | null = rpcURL
-    ? new ethers.providers.StaticJsonRpcProvider(rpcURL)
-    : null;
-
+  const [ensName, setEnsName] = useState<string | undefined>(undefined);
+  const rpcProvider: ethers.providers.Web3Provider | undefined = provider
+    ? provider
+    : undefined;
```

Comment:
I feel like this line doesn't actually do anything. If someone doesn't pass in a `provider` it will be `null`. Why do we have to make it `undefined`?

Reply:
Oh yeah i suppose you're right lol. I'll go ahead and remove that